File: MJv1BE/apps/apartment/models/customer.py
# apps/apartment/models/customer.py
from django.db import models

class Customer(models.Model):
    gender_choice = (('男', '男'), ('女', '女'))
    cno = models.CharField(db_column='Cno', primary_key=True, null=False, max_length=100)
    name = models.CharField(db_column='CName', max_length=100, null=False)
    gender = models.CharField(db_column='Gender', choices=gender_choice, max_length=20, null=False)
    mobile = models.CharField(db_column='Mobile', max_length=20, null=False)
    roomNo = models.CharField(db_column='RoomNo', max_length=20, null=False)
    checkInTime = models.DateTimeField(db_column='CheckInTime', null=False)
    checkOutTime = models.DateTimeField(db_column='CheckOutTime', null=True)
    resideTimePeriod = models.CharField(db_column='resideTimePeriod',max_length=100, null=False)
    idCardNo = models.CharField(db_column='IDCardNo', max_length=20, null=False)
    image = models.CharField(db_column='Image', max_length=100, null=True)
    balance = models.DecimalField(max_digits=10, decimal_places=2, null=True, db_column='Balance')


    class Meta:
        managed = True
        db_table = 'customer'
        app_label = 'apartment'

    def __str__(self):
        return '序号：%s\t姓名：%s\t房间号：%s\t' % (self.cno, self.name, self.roomNo)

File: MJv1FE/src/types/Customer.ts
export interface Customer {
    cno: string
    name: string
    idCardNo: string
    mobile: string
    gender: string
    roomNo: string
    checkOutTime: Date | null
    checkInTime: Date | null
    image: string
    imageUrl: string
    balance: number | null  // 修改这一行，从 number 改为 number | null
    resideTimePeriod: string[]
    status: string
}
// 客户模型接口定义
export interface CustomerInterface {
    cno: string; // 客户编号（主键）
    name: string; // 客户姓名
    idCardNo: string; // 身份证号
    mobile: string; // 手机号码
    gender: string; // 性别
    roomNo: string; // 房间号
    checkInTime: Date | null; // 入住时间
    checkOutTime: Date | null; // 退房时间（可为null）
    image: string
    imageUrl: string
    resideTimePeriod: string; // 居住时间段
    balance: number | null;
    status: string; // 状态
}

export enum CustomerDialogStatus {
    NONE = 'none',           // 无状态
    EDIT = 'edit',           // 修改状态
    VIEW = 'view',           // 查看状态
    ADD = 'add',             // 添加状态
    RENEWAL = 'renewal',      // 续租状态
    CHECKINFORSERVED = 'checkInForServed', // 预订顾客办理入住
    CUSTOMERRSERVED = 'customerServed', // 顾客办理入住
}